### PR TITLE
save loss to disk

### PR DESCRIPTION
**Pull request by jguoxu**
* refactor training loop
* save loss to disk

### CODE REVIEW

File: task.py [hunk 1]
```diff
@@ -124,31 +127,32 @@ def train():
...
-                    else:
-                        output_predict(logits_val, images_val, "data/predict_%05d_%05d" % (step, i))
-                index += 1
-
-            # save parameters every 5 epoch.
-            if step % 2000 == 0 or (step * 1) == MAX_STEPS:
+
+            # save a debug image every 1000 epoch
+            # result in 10000 image
+            if i % 1000 == 0:
```
Comment (9yoi) [inline task.py]: did you intend to save it for step instead of i?

Comment (jguoxu) [reply task.py]: Good catch, updated
